test/results_test.dart

```diff
@@ -662,4 +662,34 @@ Future<void> main([List<String>? args]) async {
       if (count > 1) fail('Should only receive one event');
     }
   });
+
+  test('RealmResults.isValid', () {
+    final config = Configuration.local([Team.schema, Person.schema]);
+    final realm = getRealm(config);
+
+    final alice = Person('Alice');
+    final bob = Person('Bob');
+    final carol = Person('Carol');
+    final dan = Person('Dan');
+
+    final team = realm.write(() {
+      return realm.add(Team('Class of 92', players: [alice, bob, carol, dan]));
+    });
+
+    final players = team.players;
+    final playersAsResults = team.players.asResults();
+
+    expect(players.isValid, isTrue);
+    expect(playersAsResults.isValid, isTrue);
+    expect(playersAsResults, [alice, bob, carol, dan]);
+
+    realm.write(() => realm.delete(team));
+
+    expect(team.isValid, isFalse); // dead object
+    expect(players.isValid, isFalse); // parent is dead
+    expect(() => players.isEmpty, throwsException);
+    expect(playersAsResults.isValid, isFalse);
+    expect(() => playersAsResults, throwsException);
+    expect(() => playersAsResults.freeze(), throwsException);
```

Comment:
Should we mark this with a TODO that we will be fixing it not to crash but return an empty list?

Reply:
Core reversed their thinking on this and decided it should throw, but it is consistent with `isValid` returning `false`

Reply:
And now they reversed their thinking again..